pkg/runtime/k0s/delete_masters.go

```diff
@@ -0,0 +1,124 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k0s
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	RemoteCleanMasterOrNode = `if which k0s; then k0s stop && k0s reset --config %s --cri-socket %s;fi && \
+rm -rf /etc/k0s/
+`
+	RemoveKubeConfig          = "rm -rf /usr/bin/kube* && rm -rf ~/.kube/"
+	RemoveK0sBin              = "rm -rf /usr/bin/k0s"
+	RemoteRemoveEtcHost       = "sed -i \"/%s/d\" /etc/hosts"
+	RemoteRemoveRegistryCerts = "rm -rf " + DockerCertDir + "/%s*"
+	KubeDeleteNode            = "kubectl delete node %s"
+)
+
+func (k *Runtime) deleteMasters(masters []net.IP) error {
+	if len(masters) == 0 {
+		return nil
+	}
+	eg, _ := errgroup.WithContext(context.Background())
+	for _, master := range masters {
+		master := master
+		eg.Go(func() error {
+			master := master
+			logrus.Infof("Start to delete master %s", master)
+			if err := k.deleteMaster(master); err != nil {
+				logrus.Errorf("failed to delete master %s: %v", master, err)
+			} else {
+				logrus.Infof("Succeeded in deleting master %s", master)
+			}
+			return nil
+		})
+	}
+	return eg.Wait()
+}
+
+func (k *Runtime) deleteMaster(master net.IP) error {
```

Comment:
This is a detailed deep implementation of delete a master node. I think we should explain clearly what is the **STEPS** to delete a master by comments. If we understand the steps and then the code review could be quite easy. Actually we should review the design firstly, and code secondly.  No design there, and we can hardly do the code review. Do you think so ?

Reply:
Yep! when I read kubernetes runtime module, I have the same feel. which files to delete? and what logic to follow. I so sorry to made this, fix this ASAP.